crates/indexer/src/models/v2_objects.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+// This is required because a diesel macro makes clippy sad
+#![allow(clippy::extra_unused_lifetimes)]
+#![allow(clippy::unused_unit)]
+
+use crate::{
+    models::move_resources::MoveResource,
+    schema::{current_objects, objects},
+    util::standardize_address,
+};
+use aptos_api_types::{DeleteResource, Transaction, WriteResource, WriteSetChange};
+use bigdecimal::BigDecimal;
+use field_count::FieldCount;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+use super::token_models::v2_token_utils::V2TokenResource;
+
+#[derive(Debug, Deserialize, FieldCount, Identifiable, Insertable, Serialize)]
+#[diesel(primary_key(transaction_version, write_set_change_index))]
+#[diesel(table_name = objects)]
+pub struct Object {
+    pub transaction_version: i64,
+    pub write_set_change_index: i64,
+    pub object_address: String,
+    pub owner_address: Option<String>,
+    pub state_key_hash: String,
+    pub guid_creation_num: Option<BigDecimal>,
+    pub allow_ungated_transfer: Option<bool>,
+    pub is_deleted: bool,
+}
+
+#[derive(Debug, Deserialize, FieldCount, Identifiable, Insertable, Serialize)]
+#[diesel(primary_key(object_address))]
+#[diesel(table_name = current_objects)]
+pub struct CurrentObject {
+    pub object_address: String,
+    pub owner_address: Option<String>,
+    pub state_key_hash: String,
+    pub allow_ungated_transfer: Option<bool>,
+    pub last_guid_creation_num: Option<BigDecimal>,
+    pub last_transaction_version: i64,
+    pub is_deleted: bool,
+}
+
+impl Object {
+    /// Only parsing 0x1 ObjectCore from transactions
+    pub fn from_transaction(
+        transaction: &Transaction,
+    ) -> (Vec<Self>, HashMap<String, CurrentObject>) {
+        if let Transaction::UserTransaction(user_txn) = transaction {
```

Comment:
FYI, today we don't handle GenesisTransactions, but you may need to handle them in the future.

A GenesisTransaction is a writeset transaction... and isn't always at genesis

Reply:
objects aren't in genesis right? I do handle genesis transactions just case by case.

Reply:
Yeah this is likely lower in priority as we don't use GenesisTxs for some sort of checkpointing the network yet. And the first Genesis tx wouldn't have objects.